.github/workflows/publish-artifacts.yml

```diff
@@ -0,0 +1,187 @@
+name: Publish artifacts to ORAS
+
+on:
+  push:
+    branches:
+    - main
+
+env:
+  RUST_TOOLCHAIN: 1.76.0
+
+jobs:
+  publish-aa:
+    permissions:
+      contents: read
+      packages: write
+      id-token: write
+    strategy:
+      matrix:
+        tee:
+        - none
+        - amd
+        - az-cvm-vtpm
+        - tdx
+        - se
+        - cca
+        arch:
+        - x86_64
+        - s390x
+        exclude:
+        - tee: amd
+          arch: s390x
+        - tee: az-cvm-vtpm
+          arch: s390x
+        - tee: tdx
+          arch: s390x
+        - tee: se
+          arch: x86_64
+        - tee: cca
+          arch: s390x
+        include:
+        - tee: none
+          arch: x86_64
+          libc: musl
+        - tee: none
+          arch: s390x
+          libc: gnu
+        - tee: amd
+          arch: x86_64
+          libc: musl
+        - tee: az-cvm-vtpm
+          arch: x86_64
+          libc: gnu
+        - tee: tdx
+          arch: x86_64
+          libc: gnu
+        - tee: se
+          arch: s390x
+          libc: gnu
+        - tee: cca
+          arch: x86_64
+          libc: musl
```

Comment:
Optional suggestion, that I've used when the matrix combinations get messy and difficult (for me) to read:
```
        platforms: [
          { tee: none, arch: x86_64, libc: musl},
          { tee: amd, arch: x86_64, libc: musl},
          { tee: az-cvm-vtpm, arch: x86_64, libc: musl},
          { tee: tdx, arch: x86_64, libc: musl},
          { tee: cca, arch: x86_64, libc: musl},
          { tee: none, arch: s390x, libc: gnu},
          { tee: se, arch: s390x, libc: gnu},
        ]
```
then you can use `matrix.platforms.arch/tee/libc` in the references

Reply:
interesting, can you extend the example to a `strategy` block?

Reply:
great, that is way more legible!